WTWSMS/common/buildings/WtWSMS_historical_temple.txt

```diff
@@ -0,0 +1,183 @@
+temple = {
+
+	tp_hagiasophia = {  # Hagia Sophia
+		potential = {
+			FROMFROM = { title = b_hagiasophia }
+			FROM = { had_global_flag = { flag = hagia_sophia_built days = 0 } }
+		}
```

Comment:
OK, I didn't test it, just that always = no (in potential with maybe a custom tooltip if that exists on buildings, trigger can have the barony) ensures that it doesn't work in any circumstance.